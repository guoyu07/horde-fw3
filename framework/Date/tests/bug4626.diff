001- MP1 1+ SA #0
001+ MP 1+ SA #0
002- FREQ=MONTHLY;INTERVAL=1;BYDAY=1SA
002+ FREQ=MONTHLY;INTERVAL=;BYDAY=1SA