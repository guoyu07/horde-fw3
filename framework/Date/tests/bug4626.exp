MP1 1+ SA #0
FREQ=MONTHLY;INTERVAL=1;BYDAY=1SA