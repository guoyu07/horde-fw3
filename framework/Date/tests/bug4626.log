
---- EXPECTED OUTPUT
MP1 1+ SA #0
FREQ=MONTHLY;INTERVAL=1;BYDAY=1SA
---- ACTUAL OUTPUT
MP 1+ SA #0
FREQ=MONTHLY;INTERVAL=;BYDAY=1SA
---- FAILED
