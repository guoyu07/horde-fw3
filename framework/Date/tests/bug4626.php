<?php

require 'Horde/Date/Recurrence.php';
require 'Horde/iCalendar.php';

$rrule = new Horde_Date_Recurrence('2008-04-05 00:00:00');
$rrule->setRecurType(HORDE_DATE_RECUR_MONTHLY_WEEKDAY);
$rrule->setRecurOnDay(HORDE_DATE_MASK_SATURDAY);

echo $rrule->toRRule10(new Horde_iCalendar()) . "\n";
echo $rrule->toRRule20(new Horde_iCalendar()) . "\n";

?>
