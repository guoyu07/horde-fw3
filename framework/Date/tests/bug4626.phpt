--TEST--
Bug #4626: Wrong monthly-by-day recurrence rule generation
--FILE--
<?php

require 'Horde/Date/Recurrence.php';
require 'Horde/iCalendar.php';

$rrule = new Horde_Date_Recurrence('2008-04-05 00:00:00');
$rrule->setRecurType(HORDE_DATE_RECUR_MONTHLY_WEEKDAY);
$rrule->setRecurOnDay(HORDE_DATE_MASK_SATURDAY);

echo $rrule->toRRule10(new Horde_iCalendar()) . "\n";
echo $rrule->toRRule20(new Horde_iCalendar()) . "\n";

?>
--EXPECT--
MP1 1+ SA #0
FREQ=MONTHLY;INTERVAL=1;BYDAY=1SA
