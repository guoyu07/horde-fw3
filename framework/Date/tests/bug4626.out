MP 1+ SA #0
FREQ=MONTHLY;INTERVAL=;BYDAY=1SA